server/src/KeycloakCrudService.ts

```diff
@@ -0,0 +1,102 @@
+import { CRUDService, GraphbackDataProvider, GraphbackPubSub } from "@graphback/runtime";
+import { GraphbackOperationType } from '@graphback/core';
+import { GraphQLObjectType } from 'graphql'
+
+export type CrudOperationAuthRules = {
+  roles: string[]
+}
+
+
+export type CrudServiceAuthRules = {
+  create: CrudOperationAuthRules,
+  read: CrudOperationAuthRules,
+  update: CrudOperationAuthRules,
+  delete: CrudOperationAuthRules,
+}
+
+export type KeycloakCrudServiceOptions = {
+  modelType: GraphQLObjectType,
+  db: GraphbackDataProvider,
+  subscriptionConfig: GraphbackPubSub,
+  authRules: CrudServiceAuthRules,
+}
+
+/**
+ * This custom CRUD Service shows another potential way to add auth
+ * 
+ * This is actually quite nice and clean but it does not allow for field level auth.
+ * It's still a possibility that we could go with though!
+ */
+export class KeycloakCrudService<T = any> extends CRUDService {
+
+  private authRules: CrudServiceAuthRules;
+
+  constructor({ modelType, db, subscriptionConfig, authRules }: KeycloakCrudServiceOptions) {
+    super(modelType, db, subscriptionConfig);
+    this.authRules = authRules
+  }
+
```

Comment:
This file shows how we could potentially use a CRUD service to implement auth. I was investigating this before I decided to go with the resolver wrapping approach. It's still a possibility.